camel-core/src/main/java/org/apache/camel/util/jsse/BaseSSLContextParameters.java

```diff
@@ -515,7 +521,11 @@ public SSLServerSocketFactory configure(SSLServerSocketFactory factory) {
             
             @Override
             public SSLSocket configure(SSLSocket socket) {
-                
+
+                SSLParameters sslParameters = socket.getSSLParameters();
+                sslParameters.setServerNames(getSNIHostNames());
```

Comment:
I wonder if you should only call this setServiceNames if the host names are not empty (eg there is some values)